WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/ViewsAndVisitorsUseCase.kt

```diff
@@ -0,0 +1,277 @@
+                            selectedItem.period
+                    )
+            )
+            items.add(
+                    viewsAndVisitorsMapper.buildInformation()
+            )
+            items.add(
+                    viewsAndVisitorsMapper.buildChips(
+                            selectedItem,
+                            this::onColumnSelected,
+                            uiState.selectedPosition
+                    )
+            )
+        } else {
+            selectedDateProvider.onDateLoadingFailed(statsGranularity)
+            AppLog.e(T.STATS, "There is no data to be shown in the overview block")
+        }
+        return items
+    }
+
+    private fun buildTitle() = TitleWithMore(string.stats_insights_views_and_visitors, moreAction = this::onMenuClick)
+
+    private fun onMenuClick(view: View) {
+        // TODO: Connect this to second level navigation later
+        popupMenuHandler.onMenuClick(view, type)
+    }
+
+    private fun onBarSelected(period: String?) {
+        analyticsTracker.trackGranular(
+                AnalyticsTracker.Stat.STATS_OVERVIEW_BAR_CHART_TAPPED,
+                statsGranularity
+        )
+        if (period != null && period != "empty") {
+            val selectedDate = statsDateFormatter.parseStatsDate(statsGranularity, period)
+            selectedDateProvider.selectDate(
+                    selectedDate,
+                    statsGranularity
+            )
+        }
+    }
+
+    private fun onColumnSelected(position: Int) {
+        analyticsTracker.trackGranular(
+                AnalyticsTracker.Stat.STATS_OVERVIEW_TYPE_TAPPED,
+                statsGranularity
+        )
+        updateUiState { it.copy(selectedPosition = position) }
+    }
+
+    private fun onBarChartDrawn(visibleBarCount: Int) {
+        updateUiState { it.copy(visibleBarCount = visibleBarCount) }
+    }
+
+    data class UiState(val selectedPosition: Int = 0, val visibleBarCount: Int? = null)
+
+    class ViewsAndVisitorsUseCaseFactory
+    @Inject constructor(
+        @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
+        @Named(BG_THREAD) private val backgroundDispatcher: CoroutineDispatcher,
+        private val statsSiteProvider: StatsSiteProvider,
+        private val selectedDateProvider: SelectedDateProvider,
+        private val statsDateFormatter: StatsDateFormatter,
+        private val viewsAndVisitorsMapper: ViewsAndVisitorsMapper,
+        private val popupMenuHandler: ItemPopupMenuHandler,
+        private val visitsAndViewsStore: VisitsAndViewsStore,
+        private val analyticsTracker: AnalyticsTrackerWrapper,
+        private val statsWidgetUpdaters: StatsWidgetUpdaters,
+        private val localeManagerWrapper: LocaleManagerWrapper,
+        private val resourceProvider: ResourceProvider
+    ) : InsightUseCaseFactory {
+        override fun build(useCaseMode: UseCaseMode) =
+                ViewsAndVisitorsUseCase(
+                        WEEKS,
```

Comment:
Granularity is WEEKS here, but in the design, it looks like granularity is DAYS. I guess we don't need to focus on this now since we'll focus the chart in another PR.

Reply:
Although X axis is showing days in a week. The design is actually comparing this week with previous week.  We may need to change this when doing graph based on whatever fits best.